homeassistant/components/sensibo/binary_sensor.py

```diff
@@ -122,32 +122,50 @@ async def async_setup_entry(
 
     coordinator = entry.runtime_data
 
-    entities: list[SensiboMotionSensor | SensiboDeviceSensor] = []
-
-    for device_id, device_data in coordinator.data.parsed.items():
-        if device_data.motion_sensors:
+    added_devices: set[str] = set()
+
+    def _add_remove_devices() -> None:
+        """Handle additions of devices and sensors."""
+
+        entities: list[SensiboMotionSensor | SensiboDeviceSensor] = []
+        _added_devices = added_devices.copy()
+
+        for device_id in _added_devices:
+            if device_id not in coordinator.previous_devices:
+                added_devices.discard(device_id)
+
+        for device_id, device_data in coordinator.data.parsed.items():
+            if device_data.motion_sensors:
+                for sensor_id, sensor_data in device_data.motion_sensors.items():
+                    if sensor_id in added_devices:
+                        continue
+                    added_devices.add(sensor_id)
+                    entities.extend(
+                        SensiboMotionSensor(
+                            coordinator, device_id, sensor_id, sensor_data, description
+                        )
+                        for description in MOTION_SENSOR_TYPES
+                    )
+
+            if device_id in added_devices:
+                continue
```

Comment:
So we store the added devices in a set, a set is really nice with subtracting, so why don't we use this to quickly get if there are entities that require change?

Reply:
I made some changes now which I think is in line with also what you were aiming for (and it also looks cleaner I think).
I only made it for `binary_sensor` (besides modifying the test further). Let me know what you think and then I'll change the rest of platforms.